### PR TITLE
Capture and report test output for failures

### PR DESCRIPTION
**Pull request by swoogles**
Let the bike-shedding around the output styling begin!
Currently it looks like this
Screen Shot 2022-08-13 at 10 17 34 PM
:

### CODE REVIEW

File: test/shared/src/main/scala/zio/test/render/TestRenderer.scala [hunk 1]
```diff
@@ -132,6 +146,13 @@ trait TestRenderer {
     )
   }
 
+  def renderOutput(output: Chunk[String]): Message =
+    Message(
+      Line.fromString("          Output Produced by Test         ".red.underlined, 2) +:
+        output.map(s => Line.fromString("| ".red + s.yellow, 2)) :+
+        Line.fromString("==========================================\n".red, 2)
```
Comment (hmemcpy) [inline test/shared/src/main/scala/zio/test/render/TestRenderer.scala]: I would prefer this to be in the ConsoleRenderer instead.

**Comment by hmemcpy** [comment]
One comment, I think `TestRenderer` should keep format-agnostic structures. Other than that LGTM!

File: test/shared/src/main/scala/zio/test/package.scala [hunk 2]
```diff
@@ -50,7 +50,7 @@ import scala.language.implicitConversions
  * }}}
  */
 package object test extends CompileVariants {
-  type TestEnvironment = Annotations with Live with Sized with TestConfig
+  type TestEnvironment = Annotations with Live with Sized with TestConfig with TestConsole
```
Comment (adamgfraser) [inline test/shared/src/main/scala/zio/test/package.scala]: This change is not necessary.

Comment (swoogles) [reply test/shared/src/main/scala/zio/test/package.scala]: Good catch,  vestige from one of the earlier approaches. Removed.

File: test/shared/src/main/scala/zio/test/render/TestRenderer.scala (hunk 1 above)
Comment (swoogles) [reply test/shared/src/main/scala/zio/test/render/TestRenderer.scala]: Totally! Done.

**Review by adamgfraser — APPROVED** [review]

**Pull request merged**